stacks_queues/queue.py

```diff
@@ -15,47 +19,95 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+    
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.front == -1:
+            self.front = 0
+            self.rear = 0
+        elif self.size == self.buffer_size:
+            raise QueueFullException('Queue is full')
+        
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1)% self.buffer_size
+        self.size += 1 
 
+    
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
+
         """
-        pass
+
+        #Check if Queue is empty and if so raise Exception
+        # Get the element from the front 
+        # Front will go to the next index in list 
+        # Change size 
+        # Check if the list is empty 
+
+        # deque takes element off the queue, and return the element (Min's comment)
+        
+        if self.size ==0:
+            raise QueueFullException('Queue is empty')
+
+        #in order to deque, you must set self.front to self.front + 1% self.buffer_size
+        deque = self.store[self.front] 
+        self.front = (self.front +1) % self.buffer_size
+        self.size -= 1 
+        return(deque)
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if self.store[self.front]:
```

Comment:
🤔 Since you do not reassign `None` as the value of whichever elements you dequeue in your `dequeue` implementation, this conditional may evaluate to `True` even when the queue is empty. I would suggest checking if your queue is empty to determine whether or not you should return the front element instead.

```suggestion
        if self.empty():
```